client/api/InnerClient.cs

```diff
@@ -253,65 +253,50 @@ private void OnNotifyFlagsLoaded(IList<string> identifiers)
 
         public bool BoolVariation(string key, Target target, bool defaultValue)
         {
+            if (!SdkInitialized) return LogAndReturnDefault(FeatureConfigKind.Boolean, key, target, defaultValue);
+
             try
             {
-                if (SdkInitialized) return evaluator.BoolVariation(key, target, defaultValue);
-
-                logger.LogWarning(
-                    "SDK not initialized, returning default variation for {Flag}", key);
-                LogEvaluationFailureError(FeatureConfigKind.Boolean, key, target, defaultValue.ToString());
-                return defaultValue;
+                return evaluator.BoolVariation(key, target, defaultValue);
             }
-
             catch (InvalidCacheStateException ex)
             {
                 logger.LogWarning(ex,
                     "Invalid cache state detected when evaluating boolean variation for flag {Key}, refreshing cache and retrying evaluation ",
                     key);
 
-                // Attempt to refresh cache
                 var result = polling.RefreshFlagsAndSegments(TimeSpan.FromMilliseconds(2000));
 
-                // If the refresh has failed or exceeded the timout, return default variation
                 if (result != RefreshOutcome.Success)
                 {
                     logger.LogError(ex,
                         "Refreshing cache for boolean variation for flag {Key} failed, returning default variation ",
                         key);
-
-                    LogEvaluationFailureError(FeatureConfigKind.Boolean, key, target, defaultValue.ToString());
-                    return defaultValue;
+                    return LogAndReturnDefault(FeatureConfigKind.Boolean, key, target, defaultValue);
                 }
 
                 try
                 {
                     return evaluator.BoolVariation(key, target, defaultValue);
                 }
-                
                 catch (InvalidCacheStateException)
                 {
                     logger.LogWarning(
-                        "SDK not initialized, returning default variation for {Flag}", key);
-
-                    LogEvaluationFailureError(FeatureConfigKind.Boolean, key, target, defaultValue.ToString());
-                    return defaultValue;
+                        "Attempted re-evaluation of boolean variation for flag {Key} after refreshing cache failed due to invalid cache state, returning default variation",
+                        key);
+                    return LogAndReturnDefault(FeatureConfigKind.Boolean, key, target, defaultValue);
                 }
             }
         }
 
         public string StringVariation(string key, Target target, string defaultValue)
         {
+            if (!SdkInitialized) return LogAndReturnDefault(FeatureConfigKind.String, key, target, defaultValue);
```

Comment:
arent all these methods the same skeleton, just with different default value and return types?

Reply:
Pretty much.

Reply:
(And for another PR. But I like your thinking 👍 )